GenderPayGap.WebUI/Controllers/Account/AccountCreationController.cs

```diff
@@ -193,6 +204,32 @@ private User CreateNewUser(CreateUserAccountViewModel viewModel)
             return user;
         }
 
+        private User UpdateRetiredUser(CreateUserAccountViewModel viewModel, User retiredUser)
+        {
+            // If user creates a new account with same email address as retired account,
+            // reuse the old account to avoid duplicates, update all details except created date and email
+            var user = retiredUser;
+            user.Modified = VirtualDateTime.Now;
+            user.Firstname = viewModel.FirstName;
+            user.Lastname = viewModel.LastName;
+            user.JobTitle = viewModel.JobTitle;
+            user.AllowContact = viewModel.AllowContact;
+            user.SendUpdates = viewModel.SendUpdates;
+
+            byte[] salt = Crypto.GetSalt();
+            user.Salt = Convert.ToBase64String(salt);
+            user.PasswordHash = Crypto.GetPBKDF2(viewModel.Password, salt);
+            user.HashingAlgorithm = HashingAlgorithm.PBKDF2;
+
+            user.EmailVerifySendDate = null;
+            user.EmailVerifiedDate = null;
+            user.EmailVerifyHash = null;
+            user.SetStatus(UserStatuses.New, user,  "Retired user account has been reactivated");
+            user.Status = UserStatuses.New;
+
+            return user;
+        }
```

Comment:
these 2 methods look very similar. I suggest we refactor them to remove the duplicated code, what do you think?

Reply:
Yeah, that seems sensible!